### ISSUE TITLE
Rare "failed to init node with kubeadm" on Kind v0.7.0

### ISSUE THREAD

**Issue by howardjohn** [kubernetes-sigs/kind#1290]
**What happened**: Occasionally find errors like https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20244/integ-telemetry-k8s-tests_istio/7089 failing to start up the kind cluster. I have seen similar errors before, but its very rare, anecdotally I see one a week or so across 1000s of tests. (meta comment: is there a good way to grep across all test logs in GCS? I cannot find one. I used to download them locally but ran out of disk space).

**What you expected to happen**: ideally no errors

**How to reproduce it (as minimally and precisely as possible)**: I cannot repro

**Anything else we need to know?**:
I don't really expect too much here. As I see errors in the future I'll add more context to maybe root cause this. Right now I am not too concerned with this as its extremely uncommon, mostly just opening this to track or if it helps find some issue

**Environment:**
- kind version: (use `kind version`): v0.7.0
- Kubernetes version: (use `kubectl version`): v1.17.0
- Docker version: (use `docker info`): 19.03.2
- OS (e.g. from `/etc/os-release`): linux

**Comment by BenTheElder** [comment]
> (meta comment: is there a good way to grep across all test logs in GCS? I cannot find one. I used to download them locally but ran out of disk space).

there isn't AFAIK. for kubernetes someone hacked up a dashboard to allow searching test failures at some point but only for junit results and not for pod-utils jobs IIRC ... :|

**Comment by BenTheElder** [comment]
> https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20244/integ-telemetry-k8s-tests_istio/7089

This looks like the API server had trouble starting up in a reasonable time frame, kubeadm gave up after about 4 minutes of waiting on the healthz. We could probably ask kubeadm to allow a longer wait but I'm not sure that's desirable.

The results do have logs from the API server, I don't see anything obvious in them suggesting a clear KIND bug, it kinda looks like things were just running to slowly.

Currently my plan for this is to attempt to tweak our etcd configuration to make it lighter on I/O as that's my main suspect other than just ... Kubernetes. I had a breif chat with our team about possible options for this last week but we're not shipping anything just yet.

It could also be prow giving us issues with noisy shared hosts. I'm not seeing this in our prow running kind at HEAD though, we actually haven't had a cluster bringup flake in a long time (plenty of e2e flakes though...).

Currently we run k8s-prow-builds with 108 nodes of:
>Image type: Ubuntu with Containerd (ubuntu_containerd)
>Machine type: n1-highmem-8
>Boot disk type: SSD persistent disk
>Boot disk size (per node): 250 GB

**Comment by howardjohn** [comment]
We run n1-standard-16 with 250gb SSD on COS. Our test load is pretty high right now so the nodes are pretty packed, seems very likely to just be noisy neighbor type issues. I am pretty happy with how stable things are now, this is definitely not a high priority (to me)

**Comment by BenTheElder** [comment]
ack, thanks!

**Comment by howardjohn** [comment]
This one is a bit different but I figured I'd add it here https://prow.istio.io/view/gcs/istio-prow/logs/integ-security-k8s-tests_istio_postsubmit/1293

Same story as before, we get these extremely rare and seemingly random. This one feels really different though, seems like the container doesn't run correctly

**Comment by neolit123** [comment]
can you enable more verbosity for the `kind` commands in the output of your CI?

side note, in the kubeadm CI (uses kinder, based on kind) we are seeing a very odd and fairly rare failure in `docker cp` when calls copying from host to container. my only explanation is a docker bug at this point.

**Comment by howardjohn** [comment]
We pass -v9 so I assume that's as verbose as it gets?

**Comment by neolit123** [comment]
`--loglevel=debug` used to add information about the `docker` calls.

> Error response from daemon: Container ad88ad13645845281b0f9fac515e3b9925f00c3f1cc9c8041ea4b69c0eabb1de is not running

you might have too look at the logs for the docker daemon too.

**Comment by BenTheElder** [comment]
> side note, in the kubeadm CI (uses kinder, based on kind) we are seeing a very odd and fairly rare failure in docker cp calls when copying small YAML files from host to container. my only explanation is a docker bug at this point.

kind doesn't use `docker cp` fwiw, but I do think the nested docker daemon may flake on some streaming API calls.. :/

> Same story as before, we get these extremely rare and seemingly random. This one feels really different though, seems like the container doesn't run correctly

yeah this is my concern, I'm not sure what _we're_ doing to cause this though..

> We pass -v9 so I assume that's as verbose as it gets?

atm that's more than maximum.

**Comment by BenTheElder** [comment]
@amwat has done a few rounds of stress testing locally w/ repeated `kind export logs` (many docker calls to stream files) and we're not seeing it locally.

**Comment by cofyc** [comment]
hi, 

We encountered this issue too and found that the `sync` command in `entrypoint` script may run for a long time sometimes (high system load?).

https://github.com/kubernetes-sigs/kind/blob/c0a7803bc09961b7a6b84f48fa98fed172812320/images/base/files/usr/local/bin/entrypoint#L28-L31

Maybe we can skip this command when storage driver is not `aufs`?

example fail logs:

```
...
[2020-02-05T06:23:26.578Z] I0205 06:23:24.394769      45 round_trippers.go:438] GET https://172.17.0.3:6443/healthz?timeout=32s  in 0 milliseconds
[2020-02-05T06:23:26.578Z] [kubelet-check] It seems like the kubelet isn't running or healthy.
[2020-02-05T06:23:26.578Z] [kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get http://localhost:10248/healthz: dial tcp 127.0.0.1:10248: connect: connection refused.
[2020-02-05T06:23:26.578Z] 
[2020-02-05T06:23:26.578Z] Unfortunately, an error has occurred:
[2020-02-05T06:23:26.578Z] 	couldn't initialize a Kubernetes cluster
[2020-02-05T06:23:26.579Z] timed out waiting for the condition
[2020-02-05T06:23:26.579Z] 
[2020-02-05T06:23:26.579Z] This error is likely caused by:
[2020-02-05T06:23:26.579Z] 	- The kubelet is not running
[2020-02-05T06:23:26.579Z] 	- The kubelet is unhealthy due to a misconfiguration of the node in some way (required cgroups disabled)
[2020-02-05T06:23:26.579Z] 
[2020-02-05T06:23:26.579Z] If you are on a systemd-powered system, you can try to troubleshoot the error with the following commands:
[2020-02-05T06:23:26.579Z] 	- 'systemctl status kubelet'
[2020-02-05T06:23:26.579Z] 	- 'journalctl -xeu kubelet'
[2020-02-05T06:23:26.579Z] 
[2020-02-05T06:23:26.579Z] Additionally, a control plane component may have crashed or exited when started by the container runtime.
[2020-02-05T06:23:26.579Z] To troubleshoot, list all containers using your preferred container runtimes CLI, e.g. docker.
[2020-02-05T06:23:26.579Z] Here is one example how you may list all Kubernetes containers running in docker:
[2020-02-05T06:23:26.579Z] 	- 'docker ps -a | grep kube | grep -v pause'
[2020-02-05T06:23:26.579Z] 	Once you have found the failing container, you can inspect its logs with:
[2020-02-05T06:23:26.579Z] 	- 'docker logs CONTAINERID'
[2020-02-05T06:23:26.579Z]  ✗ Starting control-plane 🕹️
[2020-02-05T06:23:30.751Z] ERROR: failed to create cluster: failed to init node with kubeadm: command "docker exec --privileged tidb-operator-control-plane kubeadm init --ignore-preflight-errors=all --config=/kind/kubeadm.conf --skip-token-print --v=6" failed with error: exit status 1
```

**Comment by BenTheElder** [comment]
skipping the sync sounds reasonable.

**Comment by BenTheElder** [comment]
that sync has been present since v0.4.0 though, I suspect that's more of a symptom than a root cause ...

**Comment by BenTheElder** [comment]
v0.8.0 images are shipped with the sync fix.

**Comment by BenTheElder** [comment]
Hoping 0.8.0 fixed it, if not we can always cross reference this bug from a new issue with more recent failure details. As-is at best this looks like the `sync` thing causing extra slow startup sometimes, which was fixed.
